### PR TITLE
ci: various tweaks/fixes

### PR DESCRIPTION
**Pull request by Salakar**
## Description

Adds a `flutter test` job for Web testing. Additionally has various CI tweaks post merging of core nnbd changes.

## Related Issues

N/A

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/FirebaseExtended/flutterfire/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: .github/workflows/all_plugins.yaml [hunk 1]
```diff
@@ -21,13 +21,10 @@ jobs:
...
           ./.github/workflows/scripts/install-tools.sh
-          flutter --suppress-analytics pub global activate tuneup
       - name: "Bootstrap Workspace"
         run: melos bootstrap
-      - name: "Dart Analyze"
-        run: |
-          melos exec -c 3 -- \
-            tuneup check
+      - name: "Run Analyze"
+        run: melos run analyze --no-select
```
Comment (rrousselGit) [inline .github/workflows/all_plugins.yaml:27]: What's the `--no-select`?

File: .github/workflows/all_plugins.yaml [hunk 2]
```diff
@@ -84,23 +81,27 @@ jobs:
...
+        run: |
+          ./.github/workflows/scripts/install-tools.sh
+          flutter config --enable-web
+      - name: "Bootstrap Workspace"
+        run: melos bootstrap
+      # TODO This only builds examples that have a web implementation. Previously this was `aot` snapshots however this
+      # TODO has since been removed from Flutter.
+      - name: "Build Examples"
+        run: |
+          melos exec -c 1 --scope="*example*" --dir-exists="web" -- \
```
Comment (rrousselGit) [inline .github/workflows/all_plugins.yaml:103]: Won't we still risk a timeout?

File: melos.yaml [hunk 3]
```diff
@@ -113,6 +113,9 @@ scripts:
         "flutter test --no-pub --platform=chrome"
     description: Run `flutter test --platform=chrome` for a specific '*web' package.
     select-package:
+      ignore:
+        # TODO(Salakar): Package has no web tests.
```
Comment (rrousselGit) [inline melos.yaml:117]: nit but I'd add an empty test instead

To not forget to uncomment this. Happened to me before :singe_rien_vu:

**Comment by rrousselGit** [comment]
LGTM~

**Review by rrousselGit — APPROVED** [review]

File: melos.yaml (hunk 3 above)
Comment (Salakar) [reply melos.yaml:117]: Fair point, will add an empty test :)

File: .github/workflows/all_plugins.yaml (hunk 2 above)
Comment (Salakar) [reply .github/workflows/all_plugins.yaml:103]: Not for web building, takes about 4 minutes to build all current web apps (7 currently) - current timeout is set to 15 minutes and max GH action timeout is 60 minutes (something we'd easily reach with 5-10 minute each Android builds - but not for web)

File: .github/workflows/all_plugins.yaml (hunk 1 above)
Comment (Salakar) [reply .github/workflows/all_plugins.yaml:27]: Doh, not needed on this one so will remove.

But to explain what it is, any script with a `select-package` configuration in the definition in melos.yaml e.g. https://github.com/FirebaseExtended/flutterfire/blob/master/melos.yaml#L31-L36 will prompt the user to select a package to run it against.

e.g. I can run `melos run build:example_android` and by default it'll ask me to choose from a pre-filtered list of packages in the workspace to run the actual script against. I could even just do `melos run` and will be prompted to select a script, then if the script has a select package, a package.

`--no-select` disables the selection behaviour but keeps the filters, meaning it'd run against all packages matching the filter.

File: melos.yaml (hunk 3 above)
Comment (Salakar) [reply melos.yaml]: Done!

**Pull request merged**